prompt2model/dataset_transformer/prompt_based.py

```diff
@@ -29,7 +32,7 @@ class PromptBasedDatasetTransformer(DatasetTransformer):
     def __init__(
         self,
         plan_prompt_fn: Callable[
-            [str, list[dict], str], str
+            [str, list[dict], str, int], str
```

Comment:
With a complicated function interface like this, it would be good to explain what this function is (and why this particular interface is required) in the `"Args:"` section of the docstring to this `__init__ `function.

Reply:
sure, did this. also rearranged variables to make more logical sense